lib/SILGen/SILGenBuiltin.cpp

```diff
@@ -1886,6 +1887,63 @@ static ManagedValue emitBuiltinInjectEnumTag(SILGenFunction &SGF, SILLocation lo
   return ManagedValue::forObjectRValueWithoutOwnership(bi);
 }
 
+/// Find the extension on DistributedActor that defines __actorUnownedExecutor.
+static ExtensionDecl *findDistributedActorAsActorExtension(
+    ProtocolDecl *distributedActorProto) {
+  ASTContext &ctx = distributedActorProto->getASTContext();
+
+  auto name = ctx.getIdentifier("__actorUnownedExecutor");
+  auto results = distributedActorProto->lookupDirect(
+      name, SourceLoc(),
+      NominalTypeDecl::LookupDirectFlags::IncludeAttrImplements);
+  for (auto result : results) {
+    if (auto var = dyn_cast<VarDecl>(result)) {
+      return dyn_cast<ExtensionDecl>(var->getDeclContext());
+    }
+  }
+
+  return nullptr;
+}
+
+static ManagedValue emitBuiltinDistributedActorAsAnyActor(
+    SILGenFunction &SGF, SILLocation loc, SubstitutionMap subs,
+    ArrayRef<ManagedValue> args, SGFContext C) {
+  auto &ctx = SGF.getASTContext();
+  auto distributedActor = args[0];
+
+  auto builtinDecl = cast<FuncDecl>(getBuiltinValueDecl(
+      ctx, ctx.getIdentifier("distributedActorAsAnyActor")));
+  auto genericSignature = builtinDecl->getGenericSignature();
+  auto genericParam = genericSignature.getGenericParams()[0];
+
+  auto distributedActorProto = ctx.getProtocol(KnownProtocolKind::DistributedActor);
+  auto ext = findDistributedActorAsActorExtension(distributedActorProto);
+
+  // Conformance of DistributedActor to Actor.
+  auto actorProto = ctx.getProtocol(KnownProtocolKind::Actor);
+  CanType distributedActorType = distributedActor.getType().getASTType();
+  RootProtocolConformance *daAsActorConformance = ctx.getNormalConformance(
```

Comment:
Hah, I see how that works now.. sneaky, so we can construct such conformances as hoc... -- thank you, Doug.